main.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"compress/flate"
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	newrelic "github.com/newrelic/go-agent"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/hlog"
+
+	"ride-matcher/config"
+	"ride-matcher/handlers"
+	"ride-matcher/middlewares"
+)
+
+func main() {
+	// catch the signals as soon as possible
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, syscall.SIGINT)  // a.k.a ctrl+C
```

Comment:
Here it's better to use `os` package instead of `syscall` as `os` is a higher level package.
```suggestion
	signal.Notify(signalChan, os.Interrupt)  // a.k.a ctrl+C
```